openapi-generator/src/test/java/io/micronaut/openapi/generator/KotlinMicronautClientCodegenTest.java

```diff
@@ -550,4 +550,17 @@ void testOneOfWithoutDiscriminator() {
             "@JsonTypeInfo"
         );
     }
+
+    @Test
+    void testParamsWithDefaultValue() {
+
+        var codegen = new KotlinMicronautClientCodegen();
+        String outputPath = generateFiles(codegen, "src/test/resources/3_0/params-with-default-value.yml", CodegenConstants.APIS, CodegenConstants.MODELS);
+        String path = outputPath + "src/main/kotlin/org/openapitools/";
+
+        assertFileContains(path + "api/DefaultApi.kt",
+            "@PathVariable(name = \"apiVersion\", defaultValue = \"v5\") @Nullable apiVersion: BrowseSearchOrdersApiVersionParameter? = BrowseSearchOrdersApiVersionParameter.V5,",
+            "@Header(name = \"Content-Type\", defaultValue = \"application/json\") @Nullable contentType: String? = \"application/json\""
+        );
```

Comment:
The last case I was hoping for is `XFavorToken` default null to work. Is that easy to add?

```kotlin
@Header("X-Favor-Token") @Nullable xFavorToken: String? = null
```

Not sure you'd even have a defaultValue in the annotation for null since it has to be a string

Reply:
fixed